File: DevReview.Domain/Entities/ReviewRequest.cs
using System;
using System.Collections.Generic;
using DevReview.Domain.Enums;

namespace DevReview.Domain.Entities
{
    public class ReviewRequest
    {
        public Guid Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public string ProgrammingLanguage { get; set; } = string.Empty;
        public string Framework { get; set; } = string.Empty;
        public DifficultyLevel Difficulty { get; set; } = DifficultyLevel.Junior;
        public bool IsPublic { get; set; }
        public bool IsPaid { get; set; }
        public decimal Price { get; set; }
        public string Tags { get; set; } = string.Empty;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public ReviewStatus Status { get; set; } = ReviewStatus.Open;
        public Guid OwnerId { get; set; }
        public User? Owner { get; set; }
        public Guid? MentorId { get; set; }
        public User? Mentor { get; set; }
        public DateTime? ClaimedAt { get; set; }
        public DateTime? ClaimTimeout { get; set; }
        public DateTime? CompletedAt { get; set; }

        public ICollection<CodeFile> CodeFiles { get; set; } = new List<CodeFile>();
        public ICollection<Comment> Comments { get; set; } = new List<Comment>();
        public ICollection<Review> Reviews { get; set; } = new List<Review>();
    }
}

File: DevReviewFrontend/src/types/reviewRequest.ts
import type { Comment } from './comment'
import type { DifficultyLevel, ReviewStatus } from './enums'

export interface CodeFile {
  id: string
  fileName: string
  content: string
  orderIndex: number
}

export interface CodeFileInput {
  fileName: string
  content: string
  orderIndex: number
}

export interface ReviewRequest {
  id: string
  title: string
  description: string
  programmingLanguage: string
  framework: string
  difficulty: DifficultyLevel
  isPublic: boolean
  isPaid: boolean
  price: number
  tags: string[]
  status: ReviewStatus
  createdAt: string
  ownerId: string
  mentorId: string | null
}

export interface FinalReviewSummary {
  summary: string
  priorityFixes: string
  qualityScore: number
  createdAt: string
  authorRatingOfMentor: number | null
  mentorRatingOfAuthor: number | null
}

export interface ReviewRequestDetail extends ReviewRequest {
  claimedAt: string | null
  claimTimeout: string | null
  completedAt: string | null
  codeFiles: CodeFile[]
  comments: Comment[]
  finalReview: FinalReviewSummary | null
}

export interface CreateReviewRequestPayload {
  title: string
  description: string
  programmingLanguage: string
  framework: string
  difficulty: DifficultyLevel
  isPublic: boolean
  isPaid: boolean
  price: number
  tags: string[]
  codeFiles: CodeFileInput[]
}

export interface AbandonReviewPayload {
  reason: string
}

export interface FinalizeReviewPayload {
  summary: string
  priorityFixes: string
  qualityScore: number
}
